metagraph/src/graph/annotated_dbg.cpp

```diff
@@ -25,19 +25,14 @@ typedef std::pair<Label, size_t> StringCountPair;
 
 AnnotatedSequenceGraph
 ::AnnotatedSequenceGraph(std::shared_ptr<SequenceGraph> graph,
-                         std::unique_ptr<Annotator>&& annotation,
+                         std::shared_ptr<Annotator> annotation,
```

Comment:
Why would anyone create multiple instances of AnnotatedDBG with the same (graph, annotation) pairs?
You can always create one and share it across.

Reply:
The idea behind this is that when pairing an annotator with a graph that has an underlying cache, the cache update operations have to lock every time, so this slows down performance.

If there's a single annotator which is shared across multiple wrappers of the same underlying graph, then each wrapper can be assigned to a separate thread and there's no contention.

Reply:
That idea is not realized in this PR. You still used the global cache.

Reply:
This is a vestige from the old version of `DBGSuccinct::CachedView`. Originally, that class handled all of the caching, but then this was changed and some caching duties went back to `CanonicalDBG`.

Reply:
So now, this branch needs to be changed so that the wrapping to `CanonicalDBG` happens separately for each batch of sequences.

Reply:
I'm reviewing this PR, I don't look at any other branches. If there is something there necessary for this, merge them into this PR.